File: backend/app/models/resume.py
from sqlalchemy import Column, Integer, String
from sqlalchemy.orm import relationship

from app.db.database import Base


class Resume(Base):
    __tablename__ = "resumes"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    name = Column(
        String,
        nullable=False,
        default=""
    )

    email = Column(
        String,
        nullable=False,
        default=""
    )

    contact = Column(
        String,
        nullable=False,
        default=""
    )

    portfolio = Column(
        String,
        nullable=True
    )

    address = Column(
        String,
        nullable=True
    )


    education = relationship(
        "Education",
        back_populates="resume",
        cascade="all, delete-orphan"
    )

    experience = relationship(
        "Experience",
        back_populates="resume",
        cascade="all, delete-orphan"
    )

    projects = relationship(
        "Project",
        back_populates="resume",
        cascade="all, delete-orphan"
    )

    custom_sections = relationship(
        "CustomSection",
        back_populates="resume",
        cascade="all, delete-orphan"
    )

File: frontend/src/types/resume.ts
export interface Header {
    name: string;
    email: string;
    contact: string;
    portfolio?: string;
    address?: string;
}

export interface Education {
    id: number;
    instituteName: string;
    degreeName: string;
    fromDate: string;
    toDate: string;
    cgpa?: string;
}

export interface Experience {
    id: number;
    roleTitle: string;
    instituteName: string;
    fromDate: string;
    toDate: string;
    location?: string;
    description: string;
    certificateLink?: string;
}

export interface Project {
    id: number;
    projectTitle: string;
    description: string;
    codebaseLink?: string;
    demoLink?: string;
}

export interface CustomEntry {
    id: number;
    title?: string;
    description?: string;
    link?: string;
}

export interface CustomSection {
    id: number;
    title: string;
    entries: CustomEntry[];
}

export interface Resume {
    id: number;
    header: Header;
    education: Education[];
    experience: Experience[];
    projects: Project[];
    customSections: CustomSection[];
}